src/vs/base/browser/dom.ts

```diff
@@ -1788,10 +1788,21 @@ export function h(tag: string, ...args: [] | [attributes: { $: string } & DomNod
 		children = args[1];
 	}
 
-	const [tagName, className] = tag.split('.');
+	const match = SELECTOR_REGEX.exec(tag);
+
+	if (!match) {
+		throw new Error('Bad use of h');
```

Comment:
You could throw "BugIndicatingError" here.

Reply:
I didn't know about it. Cool concept! Also, its documentation says:

> Do not throw this for invalid user input.

The throw here is literally about invalid user input.